analyzers/its/sources/BlazorSample/BlazorSample/Pages/S6798/S6798.CSharpOnly.cs

```diff
@@ -0,0 +1,15 @@
+using Microsoft.JSInterop;
+
+namespace BlazorSample.Pages;
+
+public class S6798_CSharpOnly
+{
+    [JSInvokable] private void JSInvokablePrivateMethodInPartialRazorComponent() { }
```

Comment:
The current approach I have taken here is similar to what @sebastien-marichal has done in S6800 and is basically the following:
1. one C#-only component, to show that we report even when a razor file is not there
2. one razor-only component, to show that we report even when the associated CS file is not there
3. a C# + razor component, to show that code generation is happening correctly and the two `partial` are reconciled correctly

The last scenario (and the second, to some extent) is particularly important, in my opinion, since that's not the case for .NET ITs currently, due to the limitation coming from SonarLint context, and the bug from Microsoft that breaks line directive resolution when the code generation is done in the context of the IDE.

The day this bug on the Microsoft side is fixed, these ITs will break, since the actual will contain more issues than expected. I think this is a good thing because we want to be notified of the impact that an external change is going to have on our rule.

This is the reason why I would keep them. However, I agree with you that the cases on local functions in C#-only are not very relevant to the integration and could be removed. The test cases on local functions in razor files are IMHO relevant, relevant because, again, they affect code generation, which is an integration concern (between the context, namely SonarLint or the S4NET, and the analyzer).

What do you think?

Reply:
Good points. I still don't get why we have 4 cases on `S6798_CSharpOnly`, or 5 in `S6798.Partial.razor` with `@code`, `@functions` and `@` blocks. I mean, if I understand correctly, once 1 of those is raised (because MCSFT fixed the bug) they all should raise based on the UTs, right? So I guess my question is more educational: what are we testing exactly with those?

Reply:
That's a fair point.

> once 1 of those is raised (because MCSFT fixed the bug) they all should raise based on the UTs, right

That's a guess. An educated and very reasonable guess, though. Indeed, we are testing potential issues on a fix that doesn't exist yet. So YAGNI should prevail here, and I have reduced to the bare minimum the content of each of those source files. I have applied the same logic to S6800 too (@sebastien-marichal).

Bare minimum for me means: a non-compliant case (FN in the case of .NET ITs) + a case in the positive. One could argue that even the latter may be removed since it's very unlikely that any fix from Microsoft, or any scanner-related change would make a TP into a FP. I still wanted to keep it though, since it conceptually covers the other half of the "detection space".